packages/styled/src/base.js

```diff
@@ -132,27 +132,31 @@ let createStyled: CreateStyled = (tag: any, options?: StyledOptions) => {
         newProps.ref = ref
 
         const ele = React.createElement(finalTag, newProps)
+        let possiblyStyleElement = <></>
         if (!isBrowser && rules !== undefined) {
           let serializedNames = serialized.name
           let next = serialized.next
           while (next !== undefined) {
             serializedNames += ' ' + next.name
             next = next.next
           }
-          return (
-            <>
-              <style
-                {...{
-                  [`data-emotion`]: `${cache.key} ${serializedNames}`,
-                  dangerouslySetInnerHTML: { __html: rules },
-                  nonce: cache.sheet.nonce
-                }}
-              />
-              {ele}
-            </>
+          possiblyStyleElement = (
+            <style
+              {...{
+                [`data-emotion`]: `${cache.key} ${serializedNames}`,
+                dangerouslySetInnerHTML: { __html: rules },
+                nonce: cache.sheet.nonce
+              }}
+            />
           )
         }
-        return ele
+        // Need to return the same number of siblings or else `React.useId` will cause hydration mismatches.
```

Comment:
the same fix, most likely, should also be applied in those places:
- https://github.com/emotion-js/emotion/blob/516fe458058c9ec8218740472b301e935801ebbc/packages/react/src/class-names.js#L142
- https://github.com/emotion-js/emotion/blob/516fe458058c9ec8218740472b301e935801ebbc/packages/react/src/emotion-element.js#L144
- https://github.com/emotion-js/emotion/blob/516fe458058c9ec8218740472b301e935801ebbc/packages/react/src/global.js#L137 (maybe this one can be skipped as `<Global/>` renders either `null` or `<style/>`

Reply:
`Global` seems safe. Fixed class-names and emotion-element.